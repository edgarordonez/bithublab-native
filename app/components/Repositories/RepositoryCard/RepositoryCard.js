// @flow

import React from 'react';
import { StyleSheet, View, TouchableOpacity, Image, Text } from 'react-native';
import type { Repository } from '@bithublab/types';
import ASSETS from '@bithublab/assets';

type Props = {
  repository: Repository,
  handleOnPress: (url?: string) => void,
};

export const RepositoryCard = ({ repository, handleOnPress }: Props) => {
  /* eslint-disable react/jsx-no-bind */
  const { provider, name, description, url } = repository;
  const onPressCard = () => handleOnPress(url);

  return (
    <TouchableOpacity onPress={onPressCard}>
      <View style={[styles.container, shadowStyle]}>
        <Image style={styles.image} source={ASSETS.images[provider]} />
        <View style={styles.info}>
          <Text style={styles.name}>{name}</Text>
          <Text style={styles.description}>{description}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const shadowStyle = {
  shadowOpacity: 0.5,
  shadowRadius: 2,
  shadowColor: 'black',
  shadowOffset: { width: 1, height: 1 },
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    margin: 10,
    backgroundColor: '#fff',
    padding: 20,
    elevation: 2,
  },
  image: {
    width: 25,
    height: 25,
  },
  info: {
    paddingRight: 10,
    paddingLeft: 5,
  },
  name: {
    fontFamily: 'Roboto-Black',
    fontSize: 20,
    marginBottom: 5,
  },
  description: {
    fontFamily: 'Roboto-Regular',
    fontSize: 14,
  },
});
