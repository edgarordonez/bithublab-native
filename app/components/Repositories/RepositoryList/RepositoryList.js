// @flow

import React from 'react';
import { FlatList } from 'react-native';
import type { Repository } from '@bithublab/types';
import { RepositoryCard } from '@bithublab/components';

type Props = {
  repositories: Repository[],
  handleOnPress: () => void,
};

export const RepositoryList = ({ repositories, handleOnPress }: Props) => {
  /* eslint-disable react/jsx-no-bind */
  const renderCards = (item, handleOnPress) => (
    <RepositoryCard repository={item} handleOnPress={handleOnPress} />
  );
  const extractKeyRepositories = (item, index) => `${index}`;
  const renderItem = ({ item }) => renderCards(item, handleOnPress);

  return (
    <FlatList
      data={repositories}
      keyExtractor={extractKeyRepositories}
      renderItem={renderItem}
    />
  );
};
