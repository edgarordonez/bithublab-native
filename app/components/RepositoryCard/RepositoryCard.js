// @flow

import React from 'react';
import { StyleSheet, View, Image, Text } from 'react-native';

export const RepositoryCard = () => (
  <View style={[styles.container, shadowStyle]}>
    <Image
      style={styles.image}
      source={{
        uri:
          'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png',
      }}
    />
    <View style={styles.info}>
      <Text style={styles.title}>edgarordonez/d3-stencil</Text>
      <Text style={styles.description}>
        Components of graphs build with StencilJS & D3.js
      </Text>
    </View>
  </View>
);

const shadowStyle = {
  shadowOpacity: 0.5,
  shadowRadius: 2,
  shadowColor: 'black',
  shadowOffset: { width: 1, height: 1 },
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    margin: 10,
    backgroundColor: '#fff',
    padding: 20,
  },
  image: {
    width: 25,
    height: 25,
  },
  info: {
    height: 80,
    paddingRight: 10,
    paddingLeft: 5,
  },
  title: {
    fontFamily: 'Roboto-Black',
    fontSize: 20,
    marginBottom: 5,
  },
  description: {
    fontFamily: 'Roboto-Regular',
    fontSize: 14,
  },
});
