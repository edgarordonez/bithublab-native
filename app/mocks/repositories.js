// @flow
import { Repository } from '@bithublab/components';

export const REPOSITORES: Repository[] = [
  {
    provider: 'github',
    name: 'D3-Stencil',
    description: 'Components of graphs build with StencilJS & D3.js',
    url: 'https://www.npmjs.com/package/d3-stencil',
  },
  {
    provider: 'gitlab',
    name: 'D3-Stencil',
    description: 'Components of graphs build with StencilJS & D3.js',
    url: 'https://www.npmjs.com/package/d3-stencil',
  },
  {
    provider: 'bitbucket',
    name: 'D3-Stencil',
    description: 'Components of graphs build with StencilJS & D3.js',
    url: 'https://www.npmjs.com/package/d3-stencil',
  }
];
