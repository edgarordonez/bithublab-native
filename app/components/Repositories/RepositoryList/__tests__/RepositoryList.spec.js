// @noflow

import React from 'react';
import TestRenderer from 'react-test-renderer';
import { RepositoryMock } from '@bithublab/mocks';

import { RepositoryList } from '../RepositoryList';
import { RepositoryCard } from '../../RepositoryCard/RepositoryCard';

const handleOnPress = jest.fn();
const Props = {
  repositories: [RepositoryMock],
  handleOnPress,
};

describe('RepositoryList', () => {
  it('Renders without crashing', () => {
    const rendered = TestRenderer.create(<RepositoryList {...Props} />);
    expect(rendered).toMatchSnapshot();
  });

  it('Should call "handleOnPress" when the user clicks on the card', () => {
    const rendered = TestRenderer.create(<RepositoryList {...Props} />);
    const instance = rendered.root;

    instance.findByType(RepositoryCard).props.handleOnPress();

    expect(handleOnPress).toHaveBeenCalledTimes(1);
  });
});
