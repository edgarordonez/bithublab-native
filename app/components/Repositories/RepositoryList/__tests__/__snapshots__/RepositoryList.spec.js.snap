// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryList Renders without crashing 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "description": "Components of graphs build with StencilJS & D3.js",
        "name": "edgarordonez/d3-stencil",
        "provider": "github",
        "url": "https://github.com/edgarordonez/d3-stencil",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  invertStickyHeaders={undefined}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "elevation": 2,
                "flexDirection": "row",
                "margin": 10,
                "padding": 20,
              },
              Object {
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.5,
                "shadowRadius": 2,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "width": 25,
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 5,
                "paddingRight": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Roboto-Black",
                  "fontSize": 20,
                  "marginBottom": 5,
                }
              }
            >
              edgarordonez/d3-stencil
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 14,
                }
              }
            >
              Components of graphs build with StencilJS & D3.js
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
